app/models/spree/stock/availability_validator.rb

```diff
@@ -3,14 +3,17 @@ module Stock
     # Overridden from spree core to make it also check for assembly parts stock
     class AvailabilityValidator < ActiveModel::Validator
       def validate(line_item)
+        unit_count = line_item.inventory_units.size  
+        return if unit_count >= line_item.quantity
+        
```

Comment:
I don't think this will work. I think this is what's needed...

```ruby
unit_count = line_item.inventory_units.size
parts_total = line_item.parts.to_a.sum { |v| line_item.count_of(v) } * line_item.quantity

return if unit_count > parts_total
```